packages/CollapsibleCard/src/components/Header/Header.js

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { v4 as uuidv4 } from "uuid";
+import Button from "@paprika/button";
+import ArrowUpIcon from "@paprika/icon/lib/ArrowUp";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import CollapsibleCardContext from "../../CollapsibleCardContext";
+import * as sc from "./Header.styles";
+
+export default function Header(props) {
+  let resizeTimeout;
+  const I18n = useI18n();
+  const { breakpoint, onChangeIsBroken } = props;
+
+  const headerRef = React.useRef();
+  const [isBroken, setIsBroken] = React.useState(null);
+  const [collapsibleCardWidth, setCollapsibleCardWidth] = React.useState(null);
+  const context = React.useContext(CollapsibleCardContext);
+
+  React.useEffect(() => {
+    (function iife() {
+      setCollapsibleCardWidth(headerRef.current.offsetWidth);
+    })();
+  }, []);
+
+  React.useEffect(() => {
+    setIsBroken(collapsibleCardWidth < breakpoint);
+  }, [collapsibleCardWidth, breakpoint, setIsBroken]);
+
+  React.useEffect(() => {
+    onChangeIsBroken(isBroken);
+  }, [isBroken, onChangeIsBroken]);
+
+  const numberOfChildren = React.Children.count(props.children);
+  const newChildren = [];
+  React.Children.forEach(props.children, child => {
+    if (child.type.displayName !== "CollapsibleCard.Segment") {
+      console.error(
+        `CollapsibleCard.Header's children should only be of type "CollapsibleCard.Segment". You provided a child of the type "${child.type.name}".`
+      );
+    }
+
+    if (child.props.width) {
+      newChildren.push(child);
+    } else {
+      const newChild = React.cloneElement(child, {
+        key: uuidv4(),
+        width: child.props.width || 100 / numberOfChildren,
+      });
+      newChildren.push(newChild);
+    }
+  });
+
+  function handleResizeWindow() {
+    clearTimeout(resizeTimeout);
+    resizeTimeout = setTimeout(() => {
+      setCollapsibleCardWidth(headerRef.current.offsetWidth);
+    }, 200);
+  }
+
+  window.addEventListener("resize", handleResizeWindow);
```

Comment:
will need to remove this listener when unmount

Reply:
Nice catch